docs/topics/xml-rule-when-syntax.adoc

```diff
@@ -445,3 +447,91 @@ from="MyEjbRule"
 ----
 
 |====
+
+[[has-hint-syntax]]
+== <has-hint> Syntax
+
+=== Summary
+
+Use the `<has-hint>` element to test whether a file or line has a hint already associated with it. It is primarily used to prevent adding classification to a file if a hint already exists, or to implement rules for default execution when no other conditions apply. For a better understanding of the `<has-hint>` condition, see the JavaDoc for the http://windup.github.io/windup/docs/latest/javadoc/org/jboss/windup/reporting/config/HasHint.html[HasHint] class.
```

Comment:
Maybe writing `adding classification to a file` can be a little bit misleading when talking about hints.

Reply:
I've changed this to read, "...primarily used to prevent firing if a hint..." Does this read better and still capture the spirit of the sentence?

Reply:
Yes, absolutely, thanks for the changes.